org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/CodeGenerationTemplate.java

```diff
@@ -263,4 +284,14 @@ public Template createTemplate(IJavaProject project) {
 	 * Default value for catch body content
 	 */
 	public static final String CODETEMPLATE_CATCHBODY_DEFAULT = "// ${todo} Auto-generated catch block\n${exception_var}.printStackTrace();";
+
+	/**
+	 * Default value for class snippet body content
+	 */
+	public static final String CODETEMPLATE_CLASSSNIPPET_DEFAULT = "package ${package_name};\n\n/**\n * ${type_name}\n */\npublic class ${type_name} {\n\n}";
+
+	/**
+	 * Default value for interface snippet body content
+	 */
+	public static final String CODETEMPLATE_INTERFACESNIPPET_DEFAULT = "package ${package_name};\n\n/**\n * ${type_name}\n */\npublic interface ${type_name} {\n\n}";
```

Comment:
So this is a bit trickier than that.

1- You only add the package name if it's not set already, else you end up with something like:
```java
package foo.bar;

package foo.bar;

/**
 * MyClass
 */
public class MyClass {

}
```

So, for both templates, you probably want to have something like `${packageHeader}**\n * ${type_name}\n */\npublic interface/class ${type_name} {\n\n}`

and inject the resolved `${package_name};\n\n` value into the packageHeader variable *only* if necessary.